### ISSUE TITLE
[FR] Series type pixel

### ISSUE THREAD

**Issue by isentropic** [JuliaPlots/Plots.jl#2859]
Create a new series type that draws pixels. It is similar to `st=:scatter` with smallest size and no border if backend cannot draw pixels natively. PyPlot and GR are capable of this where the benefit is vastly faster plotting for huge data where line plots take too long

**Comment by isentropic** [comment]
A way to start is to see https://github.com/JuliaPlots/Plots.jl/blob/master/src/recipes.jl and understand how some of the recipes work, update natively supported [series](https://github.com/JuliaPlots/Plots.jl/blob/2fb315af0e1772c2c6caba87ccb4720983aee16b/src/recipes.jl#L13) and implement that series in backends that would support it [natively](https://github.com/JuliaPlots/Plots.jl/blob/2fb315af0e1772c2c6caba87ccb4720983aee16b/src/backends/pyplot.jl#L407). Perhaps @LakshyaKhatri  or @Diaga  would be interested in this issue

**Comment by LakshyaKhatri** [comment]
Hello @isentropic, Sorry for the late response. I was focusing on CDSAPI.jl for past few days. I am starting to look at the resources provided and open a PR asap.

**Comment by LakshyaKhatri** [comment]
Hello @isentropic, could you provide some links to API documentations for Pixel plots in pyplot? I tried to search for them and the `pyplot.imshow()` is the only thing that is comming up.

**Comment by LakshyaKhatri** [comment]
Hello @isentropic, could you provide any link to the API documentation for pixel plots in pyplot? I tried searching for it and `pyplot.imshow()` is the only thing that is coming up. The other method like you stated earlier is a scatter plot with no borders and small size.

**Comment by isentropic** [comment]
Yeah with native pyplot, you could
```
plt.plot(rand(10), ",")
```
This has nothing to do with images

**Comment by LakshyaKhatri** [comment]
I have one doubt, do I have to put the `:pixel` series type inside one of such conditions:
https://github.com/JuliaPlots/Plots.jl/blob/2fb315af0e1772c2c6caba87ccb4720983aee16b/src/backends/pyplot.jl#L407-L411

or write new condition and implement the new plotting procedure?

**Comment by isentropic** [comment]
Yes, this should be put in of the conditions, as well as supported series types in the backends.jl file. Perhaps it should only be 2d

**Comment by LakshyaKhatri** [comment]
I have implemented the recipe, but while calling the plot with `st=:pixel` it is falling back to the default scatter plot. I will debug it and open a PR then :)

**Comment by LakshyaKhatri** [comment]
Perhaps updating supported series will do the job :smile_cat: . Thanks @isentropic :raised_hands:

**Comment by LakshyaKhatri** [comment]
Hello @isentropic, can I ask what `@deps` does (it is there below all the recipes)? I found no resources on it under `RecipeBase.jl`:
https://github.com/JuliaPlots/Plots.jl/blob/5673ced5d730dc7645ca01eb753a94cadd22b2a2/src/recipes.jl#L124-L146

**Comment by isentropic** [comment]
You need to add an entry in recipes.jl I believe
